File: order-ms/src/main/java/com/espe/edu/ec/order_ms/event_listeners/OrderValidationEvent.java
package com.espe.edu.ec.order_ms.event_listeners;

import java.time.LocalDateTime;
import java.util.UUID;

import com.fasterxml.jackson.annotation.JsonFormat;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Evento de validación que viene desde otros microservicios (FleetService, etc)
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
@Builder
public class OrderValidationEvent {

    private UUID orderId;
    private String validationType; // "resources_assigned", etc
    private boolean success;
    private String errorMessage;
    private String sourceService; // "fleet-ms", "billing-ms", etc

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private LocalDateTime timestamp;
}

File: FleetService/Models/Events/OrderValidationEvent.cs
namespace FleetService.Models.Events;

public class OrderValidationEvent
{
    public Guid OrderId { get; set; }
    public string ValidationType { get; set; } // "resources_assigned", etc
    public bool Success { get; set; }
    public string? ErrorMessage { get; set; }
    public string SourceService { get; set; } = "fleet-ms";
    public string Timestamp { get; set; } = DateTime.UtcNow.ToString("yyyy-MM-ddTHH:mm:ss");
}
